collectors/python.d.plugin/python_modules/bases/FrameworkServices/PrometheusService.py

```diff
@@ -0,0 +1,43 @@
+from .UrlService import UrlService
+
+try:
+    from prometheus_client.parser import text_fd_to_metric_families
+    HAS_PROMETHEUS = True
+except ImportError:
+    HAS_PROMETHEUS = False
+
+
+def get_metric_by_name(metrics, name):
+    for metric in metrics:
+        if metric.name == name:
+            return metric
+    return None
+
+
+class PrometheusService(UrlService):
+
+    def _get_raw_data(self, url=None, manager=None):
+        raw = UrlService._get_raw_data(self, url=None, manager=None)
+
+        if not raw:
+            return None
+
+        lines = (line for line in raw.split('\n') if not _is_comment(line))
+
+        try:
+            metrics = [m for m in text_fd_to_metric_families(lines)]
+        except AttributeError as error:
+            self.error('parse error: {0}'.format(error))
+            return None
+
+        return metrics
+
+    def check(self):
+        if not HAS_PROMETHEUS:
+            self.error('python-prometheus_client package is needed to use PrometheusService')
+            return False
+        return UrlService.check(self)
+
+
+def _is_comment(line):
+    return line.startswith('#') and not line.startswith("# TYPE")
```

Comment:
Note: both comments (lines starting with `# TYPE` or `# HELP`) are meant to be a metadata information for metric lines. Official libraries doesn't provide any more "comment" lines which start from `#`.

Basically everything starting from `# HELP` is a description which we can use as a chart info.